assets/js/app.js

```diff
@@ -0,0 +1,54 @@
+function collectData() {
+    const productName = document.getElementById('productName').value
+    const price = document.getElementById('price').value
+    const productCategory = document.getElementById('productCategory').value
+    const imageUrl = document.getElementById('imageUrl').value
+    return {
+        productName: productName,
+        price:price,
+        productCategory: productCategory,
+        imageUrl:imageUrl
+    }   
+}
+
+function generateHTML(data) {
+    const newHTML = `
+        <tr>
+            <td>${data.productName}</td>
+             <td>${data.price}</td>
+              <td>${data.productCategory}</td>
+             <td><img src="${data.imageUrl}" class="color-cell" /></td>
+        </tr>    
+    `
+    return newHTML
+}
+
+function renderHTML(newHTML) {
+    const tableBody = document.getElementById('tableBody')
+    tableBody.innerHTML += newHTML
+}
+
+function validateForm(data) {
```

Comment:
1. this function is called too late
2. it doesn't add anything over the HTML validation you already have
3. bad practice to use both HTML and JS validation, you need to choose either